vcf-dumper/vcf-dumper-ws/src/main/java/uk/ac/ebi/eva/vcfdump/server/HtsgetVcfController.java

```diff
@@ -0,0 +1,213 @@
+            HttpServletRequest request,
+            HttpServletResponse response)
+            throws IllegalAccessException, IllegalOpenCGACredentialsException, InstantiationException, IOException,
+            StorageManagerException, URISyntaxException, ClassNotFoundException {
+
+        if (!VCF.equals(format)) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .body(new HtsGetError("UnsupportedFormat", "Specified format is not supported by this server"));
+        }
+
+        if (start != null && end != null && end <= start) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .body(new HtsGetError("InvalidRange", "The requested range cannot be satisfied"));
+        }
+
+        String dbName = "eva_" + species;
+        String regionStr = referenceName + ((start != null && end != null) ? ":" + start + "-" + end : "");
+        MultivaluedMap<String, String> queryParameters = new MultivaluedHashMap<>();
+        queryParameters.put(VariantDBAdaptor.REGION, Collections
+                .singletonList(regionStr));
+
+        HtsGetResponse htsGetResponse = new HtsGetResponse();
+        htsGetResponse.setFormat(VCF);
+        int blockSize = Integer.parseInt(evaProperties.getProperty("eva.htsget.blocksize"));
+
+        VariantExporterController controller = new VariantExporterController(dbName, new ArrayList<>(), evaProperties,
+                queryParameters, blockSize);
+        List<Region> regionList = controller.getRegionsForChromosome(referenceName);
+
+        htsGetResponse.constructUrls(request.getLocalName() + ":" + request.getLocalPort(), id, referenceName, species,
+                regionList);
+        return ResponseEntity.status(HttpStatus.OK).body(htsGetResponse);
+
+    }
+
+
+    @RequestMapping(value = "/headers", method = RequestMethod.GET, produces = "application/octet-stream")
+    public StreamingResponseBody getHtsgetHeaders(
+            @RequestParam(name = "species") String species,
+            @RequestParam(name = "studies") List<String> studies,
+            HttpServletResponse response)
+            throws IllegalAccessException, IllegalOpenCGACredentialsException, InstantiationException, IOException,
+            StorageManagerException, URISyntaxException, ClassNotFoundException {
+
+        String dbName = "eva_" + species;
+
+        StreamingResponseBody responseBody = getStreamingHeaderResponse(dbName, studies, evaProperties,
+                                                                        new MultivaluedHashMap<>(), response);
+
+        return responseBody;
+    }
+
+
+    @RequestMapping(value = "/block", method = RequestMethod.GET, produces = "application/octet-stream")
+    public StreamingResponseBody getHtsgetBlocks(
+            @RequestParam(name = "species") String species,
+            @RequestParam(name = "studies") List<String> studies,
+            @RequestParam(name = "chr") String chrRegion,
```

Comment:
the name "chr" is not accurate, because this param is containing chromosome but also region start and end. I would rename this param to "region", or would add two new params "start" and "end"